src/runtime/GlobalObjectBuiltinRegExp.cpp

```diff
@@ -540,6 +540,36 @@ static Value builtinRegExpMatch(ExecutionState& state, Value thisValue, size_t a
     }
 }
 
+static Value builtinRegExpMatchAll(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    if (!thisValue.isObject()) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, state.context()->staticStrings().RegExp.string(), true,
+                                       state.context()->staticStrings().toPrimitive.string(), ErrorObject::Messages::GlobalObject_ThisNotObject);
+    }
+    bool global = false;
+    bool unicode = false;
+    Object* thisObj = thisValue.asObject();
+    String* s = argv[0].toString(state);
+    String* flags = thisObj->get(state, ObjectPropertyName(state, state.context()->staticStrings().flags)).value(state, thisObj).toString(state);
+
+    Value c = thisObj->speciesConstructor(state, state.context()->globalObject()->regexp());
+    Value arguments[] = { thisObj, flags };
+    Object* matcher = Object::construct(state, c, 2, arguments);
+
+    size_t lastIndex = thisObj->get(state, ObjectPropertyName(state, state.context()->staticStrings().lastIndex)).value(state, thisObj).toNumber(state);
+    matcher->asRegExpObject()->setLastIndex(state, Value(lastIndex));
+
+    if (flags->find("g") != SIZE_MAX) {
+        global = true;
+    }
+    if (flags->find("u") != SIZE_MAX) {
+        unicode = true;
+    }
+    RegExpStringIteratorObject* iterator = new RegExpStringIteratorObject(state, global, unicode, matcher->asRegExpObject(), s);
+    return iterator;
```

Comment:
(Minor suggestion)
How about directly returning the result of new operation instead of allocating a temporal variable?